ignite/metrics/mAP.py

```diff
@@ -0,0 +1,402 @@
+            if iscrowd:
+                u = da
+            else:
+                u = da + ga - i
+            ious[m - d - 1, g] = i / u
+
+    return ious
+
+
+i_id = 0
+i_class_id = 1
+i_xmin = 2
+i_ymin = 3
+i_xmax = 4
+i_ymax = 5
+i_area = 6
+i_crowd = 7
+g_ignore = 8
+d_confidence = 8
+
+
+class AP(Metric):
+    r"""Calculates Average Precision for object detection data.
+
+    .. math::
+       \text{AP} = {1/11}\sum_{r \in {0.0, ..., 1.0}}\text{AP}_r
+
+    where \text{AP}_r represents average precision at recall value :math:`r`,
+    where :math:`r` ranges from 0.0 to 1.0.
+
+    More details can be found in `Everingham et al. 2009`__
+
+    __ https://homepages.inf.ed.ac.uk/ckiw/postscript/ijcv_voc09.pdf
+
+    Remark:
+
+        This implementation is inspired by pycocotools which can be found `here`__
+
+        __ https://github.com/cocodataset/cocoapi/
+
+    Args:
+        area_rngs: dictionary with area_name as key and a tuple of the form
+            (lower_area_range, upper_area_range) as value, where both values are floats.
+            It contains all area ranges for which AP needs to be computed.
+        max_det: Maximum detections allowed per image. This is to limit crowding and repeated
+            detections of the same object.
+        iou_thrs: A float tensor containing a set of IoU thresholds for which AP needs to be computed.
+        rec_thrs: A float tensor containing a set of values at which AP needs to be computed for averaging.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            import torch
+            from ignite.metrics import AP
+
+            gts = torch.tensor([[1,73,249.1,50.7,375.3,277.2,101928.52,0,0],
+                    [2,73,152.3,50.29,47,312.38, 4097.4,0,0]])
+            dts = torch.tensor([[1,73,176.38,50.3,463.62,312.4,144825.6,0,0.328],
+                    [2,73,355.5,111.87,210,93.75,19687.5,0,0.64]])
+
+            area_rngs = {
+                'all' : [0, float("inf")],
+            }
+            max_det = 100
+
+            iou_thrs = torch.tensor([0.5 + (i/20) for i in range(10)])
+            rec_thrs = torch.tensor([i/100 for i in range(101)])
+
+            mAP = AP(area_rngs=area_rngs, max_det=max_det, iou_thrs=iou_thrs, rec_thrs=rec_thrs)
+
+            mAP.update([[1,gts],[1,dts]])
```

Comment:
This example is not user friendly. Explain what the arguments are supposed to be.